islandora_basic_collection.module

```diff
@@ -267,60 +271,105 @@ function islandora_basic_collection_islandora_collectionCModel_islandora_view_ob
-        'js' => array(
-          'misc/form.js',
-          'misc/collapse.js',
-        ),
-      ),
       '#attributes' => array(
         'class' => array(
           'islandora-collection-metadata',
+          'islandora-collection-metadata-description',
         ),
       ),
-    );
-    $to_return['wrapper']['description'] = array(
-      '#type' => 'markup',
       '#markup' => islandora_retrieve_description_markup($object),
     );
+
+    // Add parent collections.
     $parent_collections = islandora_get_parents_from_rels_ext($object);
     if (!empty($parent_collections)) {
-      $to_return['wrapper']['collections'] = array(
+      $to_return['collections'] = array(
         '#theme_wrappers' => array('container'),
+        '#attributes' => array(
+          'class' => array(
+            'islandora-collection-metadata-in-collections',
+            'islandora-collection-metadata',
+          ),
+        ),
       );
-      $to_return['wrapper']['collections']['heading'] = array(
+      $to_return['collections']['heading'] = array(
         '#markup' => t('<h2>In collections</h2>'),
       );
-      $to_return['wrapper']['collections']['list'] = array(
+      $to_return['collections']['list'] = array(
         '#prefix' => '<ul>',
         '#suffix' => '</ul>',
       );
       foreach ($parent_collections as $parent) {
-        $to_return['wrapper']['collections']['list'][] = array(
+        $to_return['collections']['list'][] = array(
           '#markup' => l($parent->label, "islandora/object/{$parent->id}"),
           '#prefix' => '<li>',
           '#suffix' => '</li>',
         );
       }
     }
+
+    // Preserve the 'wrapper' wrapper for legacy support.
+    $to_return['wrapper'] = array(
+      '#type' => 'item',
+      '#theme_wrappers' => array('container'),
+      '#attributes' => array(
+        'class' => array(
+          'islandora-collection-metadata',
+          'islandora-collection-metadata-markup',
+        ),
+      ),
+    );
+
+    // Add the metadata fieldset markup.
     $to_return['wrapper']['metadata'] = array(
       '#type' => 'markup',
+      '#attached' => array(
+        'js' => array(
+          'misc/form.js',
+          'misc/collapse.js',
+        ),
+      ),
       '#markup' => islandora_retrieve_metadata_markup($object),
     );
   }
   return $to_return;
 }
 
+/**
+ * Implements hook_islandora_view_object_alter().
+ */
+function islandora_basic_collection_islandora_collectionCModel_islandora_view_object_alter(&$object, &$rendered) {
+  // Get the weight order from our configuration page.
+  $weight_config = variable_get('metadata_info_table_drag_attributes', NULL);
+  if (!is_null($weight_config)) {
+    foreach ($weight_config as $key => $value) {
+      if (isset($rendered[$key])) {
+        $rendered[$key]['#weight'] = $weight_config[$key]['weight'];
+        if ($weight_config[$key]['omit'] > 0) {
+          $rendered[$key]['#attributes']['class'][] = 'element-invisible';
```

Comment:
This may make more sense to be `display: none;`, the `element-invisible` class keeps things for screen readers IIRC... The intent here appears to be to turn things on and off for everybody, not just those who can physically see the elements?